File: Daily/PY/Leetcode1239-串联字符串的最大长度.py
from typing import List
class Solution:
    def maxLength(self, arr: List[str]) -> int:
        n,ans = len(arr),0
        for i in range(1 << n):
            st = [0] * 26
            flag = False
            for j in range(n):
                if (i >> j) & 1:
                    for ch in arr[j]:
                        st[ord(ch) - ord('a')] += 1
                        if st[ord(ch) - ord('a')] > 1:
                            flag = True
                            break
                if flag:break
            if not flag:ans = max(ans,sum(st))
        return ans

File: Daily/Rust/Leetcode1239-串联字符串的最大长度.rs
impl Solution {
    pub fn max_length(arr: Vec<String>) -> i32 {
        let (mut ans,n) = (0,arr.len());
        for i in 0..(1 << n){
            let mut st = vec![0;26];
            let mut flag = false;
            for j in 0..n{
                if (i >> j) & 1 == 1 {
                    for ch in arr[j].chars(){
                        st[(ch as u32- 'a' as u32) as usize] += 1;
                        if st[(ch as u32- 'a' as u32) as usize] > 1{
                            flag = true;
                            break;
                        }
                    }
                }
                if flag{
                    break;
                }
            }
            if !flag{
                ans = ans.max(st.iter().sum());
            }
        }
        ans
    }
}